src/bento/components/bento-autocomplete/1.0/types.ts

```diff
@@ -0,0 +1,53 @@
+import {Component, ComponentChildren} from '#preact/types';
+
+const filterTypes = [
+  'substring',
+  'prefix',
+  'token-prefix',
+  'fuzzy',
+  'custom',
+  'none',
+] as const;
+
+export type FilterType = typeof filterTypes[number];
+
+export function isValidFilterType(filterType: any): filterType is FilterType {
+  return filterTypes.includes(filterType);
+}
+
+export type Item = string | object;
+
+export type OnSelectData = {value: string; valueAsObject?: object};
+
+export interface BentoAutocompleteProps {
+  id?: string;
+  onError?: (message: string) => void;
```

Comment:
One question I had - a few different bento components now are using this `onError` callback, but I don't see us actually passing in this error handler from `base-element.js`. Should we still be using `user().error` or `user().warn` here?

Reply:
`console.error` should be okay. Throwing is also okay.